src/main/clojure/jarl/eval.clj

```diff
@@ -143,7 +143,7 @@
   (log/tracef "NotStmt - <%s> == <%s>" a-pos b-pos)
   (let [a (state/get-value state a-pos)
         b (state/get-value state b-pos)
-        result (= a b)]
+        result (types/rego-equal? a b)]
```

Comment:
Interesting — was this required by the changes in this PR, or just a nice unrelated change?

Reply:
There was a test case added with an equality check between `1` and `1.0`.

Reply:
Ah, right, I should have known, given how it was me who added those 😆